besu/src/test/java/org/hyperledger/besu/util/EphemeryGenesisFileTest.java

```diff
@@ -0,0 +1,135 @@
+
+  private static final GenesisConfigFile INVALID_GENESIS_JSON = fromConfig("{}");
+  private static final JsonObject INVALID_GENESIS_JSON_WITHOUT_CHAINID =
+      (new JsonObject()).put("timestamp", GENESIS_TEST_TIMESTAMP);
+
+  private static final JsonObject INVALID_GENESIS_JSON_WITHOUT_TIMESTAMP =
+      new JsonObject()
+          .put("config", (new JsonObject()).put("chainId", GENESIS_CONFIG_TEST_CHAINID));
+
+  @Test
+  public void testEphemeryWhenChainIdIsAbsent() {
+    final GenesisConfigFile config =
+        GenesisConfigFile.fromConfig(INVALID_GENESIS_JSON_WITHOUT_CHAINID.toString());
+    Optional<BigInteger> chainId = config.getConfigOptions().getChainId();
+    assertThat(chainId).isNotPresent();
+  }
+
+  @Test
+  public void testShouldDefaultTimestampToZero() {
+    final GenesisConfigFile config =
+        GenesisConfigFile.fromConfig(INVALID_GENESIS_JSON_WITHOUT_TIMESTAMP.toString());
+    assertThat(config.getTimestamp()).isZero();
+  }
+
+  @Test
+  public void testEphemeryWhenGenesisJsonIsInvalid() {
+    assertThatThrownBy(INVALID_GENESIS_JSON::getDifficulty)
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessageContaining("Invalid genesis block configuration");
+  }
+
+  @Test
+  public void testEphemeryWhenGenesisJsonIsValid() {
+    final GenesisConfigFile config = GenesisConfigFile.fromConfig(VALID_GENESIS_JSON.toString());
+    assertThat(String.valueOf(config.getTimestamp())).isNotNull();
+    assertThat(String.valueOf(config.getTimestamp())).isNotEmpty();
+    assertThat(String.valueOf(config.getConfigOptions().getChainId())).isNotNull();
+    assertThat(String.valueOf(config.getConfigOptions().getChainId())).isNotEmpty();
+  }
+
+  @Test
+  public void testEphemeryNotYetDueForUpdate() {
+    final GenesisConfigFile config = GenesisConfigFile.fromConfig(VALID_GENESIS_JSON.toString());
+    assertThat(CURRENT_TIMESTAMP).isLessThan(config.getTimestamp() + PERIOD_IN_SECONDS);
+  }
+
+  @Test
+  void testOverrideWithUpdatedChainIdAndTimeStamp() {
+    BigInteger expectedChainId =
+        BigInteger.valueOf(GENESIS_CONFIG_TEST_CHAINID)
+            .add(BigInteger.valueOf(PERIOD_SINCE_GENESIS));
+
+    long expectedGenesisTimestamp =
+        GENESIS_TEST_TIMESTAMP + (PERIOD_SINCE_GENESIS * PERIOD_IN_SECONDS);
+
+    final GenesisConfigFile config = GenesisConfigFile.fromResource("/ephemery.json");
+
+    final Map<String, String> override = new TreeMap<>(String.CASE_INSENSITIVE_ORDER);
+    override.put("chainId", String.valueOf(expectedChainId));
+    override.put("timestamp", String.valueOf(expectedGenesisTimestamp));
+
+    assertThat(config.withOverrides(override).getConfigOptions().getChainId()).isPresent();
+    assertThat(config.withOverrides(override).getTimestamp()).isNotNull();
+    assertThat(expectedChainId).isEqualTo(override.get("chainId"));
+    assertThat(String.valueOf(expectedGenesisTimestamp)).isEqualTo(override.get("timestamp"));
```

Comment:
this is asserting that the overrides has the expected values, can you also assert that the config.withOverrides has the expected values (lines 107-108 just assert it's not null)